SubsetManager alpha version

iLLiCiTiT opened the pull request:
## Changes
- implemented new tool `SubsetManager` (aplha version)
- at the moment expect that host must have implemented `list_instances` function which should return list of instances in current workfile
    - it is expected that each item is dictionary which is json serializable
    - if all these conditions are met it is possible to look at the instance data
- if host has also implemented `remove_instance` it is possible to trigger remove action
    - instance item is passed to `remove_instance` function - it's up to host's implementation how to handle of removing the instance

## TODO
- missing logging and exception handling

## Screenshot (TVPaint)
![image](https://user-images.githubusercontent.com/43494761/100370802-2181ef80-3007-11eb-91c1-0cbe1e18161d.png)

jezscha commented:
feature request as had been discussed on weekly:
- adding by adding save function in host right side of ui will be editable
- each host could have own way of formatting instance label in `list_instances()` so editorial would have `{asset}: {subset}`

iLLiCiTiT commented:
> each host could have own way of formatting instance label in list_instances() so editorial would have {asset}: {subset}

Instance data item may contain key `"label"` which is used instead of `"subset"` if is available.

> by adding save function in a host the right side of ui will be editable

Added possibility, it is possible to modify one instance at a time. Nothing will change if instance is modified and switched to different instance detail, without pushing `"Save"`.
To allow this host must have implemented `save_instances` (list of instances is passed to this function) and environment variable "`AVALON_DEVELOP_MODE`" must be set to "1", "yes", "true" or "on".
